test/res.set.js

```diff
@@ -26,24 +26,42 @@ describe('res', function(){
       });
 
       request(app)
-      .get('/')
-      .expect('X-Number', '123')
-      .expect(200, 'string', done);
+        .get('/')
+        .expect('X-Number', '123')
+        .expect(200, 'string', done);
+    })
+
+    it('should raise error if value is not a string for \"content-type\" field', function (done) {
+      var app = express();
+
+      app.use(function (req, res) {
+        try {
```

Comment:
If not wrapped in try ... catch block, it returns the whole html content and therefore does not pass the test. I don't know why this happens... Anyways, the creator of the issue has made his PR.